contracts/ColonyNetworkStaking.sol

```diff
@@ -0,0 +1,306 @@
+pragma solidity ^0.4.17;
+pragma experimental "v0.5.0";
+pragma experimental "ABIEncoderV2";
+
+import "../lib/dappsys/auth.sol";
+import "./Authority.sol";
+import "./IColony.sol";
+import "./EtherRouter.sol";
+import "./Token.sol";
+import "./ColonyNetworkStorage.sol";
+import "./IColonyNetwork.sol";
+
+
+contract ColonyNetworkStaking is ColonyNetworkStorage {
+
+  event Address(address _a);
+  event Bool(bool _b);
+  function deposit(uint _amount) public {
+    // Get CLNY address
+    Token clny = Token(IColony(_colonies["Common Colony"]).getToken());
+    uint256 networkBalance = clny.balanceOf(this);
+    // Move some over.
+    clny.transferFrom(msg.sender, this, _amount);
+    // Check it actually transferred
+    assert(clny.balanceOf(this)-networkBalance==_amount);
+    // Note who it belongs to.
+    stakedBalances[msg.sender] += _amount;
+  }
+
+  function withdraw(uint _amount) public {
+    uint256 balance = stakedBalances[msg.sender];
+    require(balance >= _amount);
+    bytes32 submittedHash;
+    (submittedHash, ) = ReputationMiningCycle(reputationMiningCycle).hasSubmitted(msg.sender);
```

Comment:
This is a public getter on hasSubmitted mapping being used meaning it doesn't go through EtherRouter?

Reply:
Correct. `ReputationMiningCycle` does not use EtherRouter, as a new `ReputationMiningCycle` is deployed every mining cycle. If we want to upgrade `ReputationMiningCycle`, we upgrade `ColonyNetworkStaking` (which deploys it) and the new `ReputationMiningCycle` contract will be used for the next cycle.